src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -244,35 +263,49 @@ private void updateWorkflowState(
                 listener.onResponse(new WorkflowResponse(workflowId));
             }, listener);
         } else {
-            // Failed deprovision
+            // Remaining resources only includes ones we tried to delete
+            List<ResourceCreated> stateIndexResources = new ArrayList<>(remainingResources);
+            // Add in those we skipped
+            stateIndexResources.addAll(deleteNotAllowed);
             flowFrameworkIndicesHandler.updateFlowFrameworkSystemIndexDoc(
                 workflowId,
                 Map.ofEntries(
                     Map.entry(STATE_FIELD, State.COMPLETED),
                     Map.entry(PROVISIONING_PROGRESS_FIELD, ProvisioningProgress.DONE),
                     Map.entry(PROVISION_END_TIME_FIELD, Instant.now().toEpochMilli()),
-                    Map.entry(RESOURCES_CREATED_FIELD, remainingResources)
+                    Map.entry(RESOURCES_CREATED_FIELD, stateIndexResources)
                 ),
                 ActionListener.wrap(updateResponse -> {
                     logger.info("updated workflow {} state to COMPLETED", workflowId);
                 }, exception -> { logger.error("Failed to update workflow {} state", workflowId, exception); })
             );
             // give user list of remaining resources
+            StringBuilder message = new StringBuilder();
+            appendResourceInfo(message, "Failed to deprovision some resources: ", remainingResources);
+            appendResourceInfo(message, "These resources require the " + ALLOW_DELETE + " parameter to deprovision: ", deleteNotAllowed);
```

Comment:
Can we merge these 2 messages?

Reply:
What do you mean by merge? They are appended onto the same string when relevant; see the test cases.

Reply:
Ah the test cleared my doubt.